Issue with file path input

> What I want to archive
> - Parse a file as input and dump the URLs into a md file
> - Check the URLs from the dumped md file as input for lychee
> 
> See my config:
> ```
>       # (1) Extract URLs
>       - name: Extract URLs
>         uses: lycheeverse/lychee-action@v2
>         with:
>           args: --dump --exclude www.w3.org -- input.xml
>           output: lychee/url_list.md
>           jobSummary: false
> 
>       # (2) Print output
>       - name: Print List of URLs
>         run: cat ./lychee/url_list.md
> 
>       # (3) Check dumped URLs from file
>       - name: Check URLs
>         uses: lycheeverse/lychee-action@v2
>         with:
>           args: --no-progress --require-https -- './lychee/url_list.md'
>           jobSummary: true
> ```
> 
> ✅ The (1) first step succeeds and the file is dumped.
> ✅ The (2) second steps succeeds and the dumped output file is found + printed.
> ❌ The (3) third step is failing and the dumped output file can not be found and used as input any more.
> ```
> Error: Cannot parse inputs from arguments
> 
> Caused by:
>     Invalid file path: ./lychee/url_list.md
> ```
> 
> Any idea? 

I'm using the same file path in (2) and (3).
I also tried `./lychee/url_list.md` without wrapping in `'`, which leads to the same error.

 _Originally posted by @tooomm in [#214](https://github.com/lycheeverse/lychee-action/issues/214#issuecomment-2692789482)_

It's printing a lot of lines... but nothing catching my attention.
I did copy the output around the ARGs and the error below:

```
...
Run # Remove the downloaded archive and any unnecessary files after installation
##[debug]/usr/bin/bash --noprofile --norc -e -o pipefail /home/runner/work/_temp/def0094a-0b1e-49d5-9261-b8c0589a6e65.sh
##[debug]Finished: run
##[debug]Evaluating: inputs.TOKEN
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'TOKEN'
##[debug]=> '***'
##[debug]Result: '***'
##[debug]Evaluating: inputs.ARGS
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'ARGS'
##[debug]=> '--no-progress --require-https --cache --max-cache-age 8h -- ./lychee/url_list.md'
##[debug]Result: '--no-progress --require-https --cache --max-cache-age 8h -- ./lychee/url_list.md'
##[debug]Evaluating: inputs.DEBUG
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'DEBUG'
##[debug]=> 'false'
##[debug]Result: 'false'
##[debug]Evaluating: inputs.FAIL
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'FAIL'
##[debug]=> 'true'
##[debug]Result: 'true'
##[debug]Evaluating: inputs.FAILIFEMPTY
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'FAILIFEMPTY'
##[debug]=> 'true'
##[debug]Result: 'true'
##[debug]Evaluating: inputs.FORMAT
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'FORMAT'
##[debug]=> 'markdown'
##[debug]Result: 'markdown'
##[debug]Evaluating: inputs.JOBSUMMARY
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'JOBSUMMARY'
##[debug]=> 'true'
##[debug]Result: 'true'
##[debug]Evaluating: inputs.CHECKBOX
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'CHECKBOX'
##[debug]=> 'true'
##[debug]Result: 'true'
##[debug]Evaluating: inputs.OUTPUT
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'OUTPUT'
##[debug]=> 'lychee/out.md'
##[debug]Result: 'lychee/out.md'
##[debug]Evaluating condition for step: 'run'
##[debug]Evaluating: success()
##[debug]Evaluating success:
##[debug]=> true
##[debug]Result: true
##[debug]Starting: run
##[debug]Loading inputs
##[debug]Evaluating: format('{0}/entrypoint.sh', github.action_path)
##[debug]Evaluating format:
##[debug]..Evaluating String:
##[debug]..=> '{0}/entrypoint.sh'
##[debug]..Evaluating Index:
##[debug]....Evaluating github:
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'action_path'
##[debug]..=> '/home/runner/work/_actions/lycheeverse/lychee-action/v2'
##[debug]=> '/home/runner/work/_actions/lycheeverse/lychee-action/v2/entrypoint.sh'
##[debug]Result: '/home/runner/work/_actions/lycheeverse/lychee-action/v2/entrypoint.sh'
##[debug]Evaluating: inputs.workingDirectory
##[debug]Evaluating Index:
##[debug]..Evaluating inputs:
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'workingDirectory'
##[debug]=> '.'
##[debug]Result: '.'
##[debug]Loading env
Run /home/runner/work/_actions/lycheeverse/lychee-action/v2/entrypoint.sh
##[debug]/usr/bin/bash --noprofile --norc -e -o pipefail /home/runner/work/_temp/8ebf6aff-f756-4b8e-9ce9-a1cec9eef789.sh
/home/runner/work/_actions/lycheeverse/lychee-action/v2/entrypoint.sh: line 48: [: : integer expression expected
Error: Cannot parse inputs from arguments

Caused by:
    Invalid file path: ./lychee/url_list.md

Error: Process completed with exit code 1.
##[debug]Set output exit_code = 1
##[debug]Finished: run
##[debug]Evaluating: steps.run-lychee.outputs.exit_code
##[debug]Evaluating Index:
##[debug]..Evaluating Index:
##[debug]....Evaluating Index:
##[debug]......Evaluating steps:
##[debug]......=> Object
##[debug]......Evaluating String:
##[debug]......=> 'run-lychee'
##[debug]....=> Object
##[debug]....Evaluating String:
##[debug]....=> 'outputs'
##[debug]..=> Object
##[debug]..Evaluating String:
##[debug]..=> 'exit_code'
##[debug]=> '1'
##[debug]Result: '1'
##[debug]Finishing: Check token art URLs
```

--- a/action.yml
+++ b/action.yml
@@ -57,16 +57,19 @@ runs:
         echo "$HOME/.local/bin" >> "$GITHUB_PATH"
         mkdir -p "$HOME/.local/bin"
       shell: bash
-    - name: Clean up existing lychee files
+    - name: Clean up existing lychee binary
       run: |
-        # Remove any existing lychee binaries or archives to prevent conflicts
+        # Remove any existing lychee binary to prevent conflicts
         rm -f "$HOME/.local/bin/lychee"
-        rm -rf lychee
-        rm -f "${{ steps.lychee-filename.outputs.filename }}"
       shell: bash
-    - name: Download and extract lychee
+    - name: Download and extract lychee in temp directory
       id: lychee-setup
       run: |
+        # Create a temporary directory for downloads and extraction
+        TEMP_DIR="${RUNNER_TEMP}/lychee-download"
+        mkdir -p "${TEMP_DIR}"
+        cd "${TEMP_DIR}"
+        
         ARCH=$(uname -m)
         # Determine filename and download URL based on version
         if [[ "${LYCHEE_VERSION}" =~ ^v0\.0|^v0\.1[0-5]\. ]]; then
@@ -89,21 +92,15 @@ runs:
         echo "Extracting ${FILENAME}"
         tar -xvzf "${FILENAME}"
 
-        # Output filename for use in later steps
-        echo "filename=${FILENAME}" >> $GITHUB_OUTPUT
+        # Output temp directory for use in later steps
+        echo "temp_dir=${TEMP_DIR}" >> $GITHUB_OUTPUT
       env:
         LYCHEE_VERSION: ${{ inputs.lycheeVersion }}
       shell: bash
     - name: Install lychee
       run: |
-        install -t "$HOME/.local/bin" -D lychee
-      shell: bash
-    - name: Clean up installation files
-      run: |
-        # Remove the downloaded archive and any unnecessary files after installation
-        rm -f "${{ steps.lychee-setup.outputs.filename }}"
-        shopt -s extglob
-        rm -f lychee!(*-bin|*-lib|*.toml)
+        # Install lychee from the temporary directory
+        install -t "$HOME/.local/bin" -D "${{ steps.lychee-setup.outputs.temp_dir }}/lychee"
       shell: bash
     - name: Run Lychee
       id: run-lychee

